resources/party.ts

```diff
@@ -151,4 +167,36 @@ export default class PartyTracker {
   nameFromId(id: string): string | undefined {
     return this.idToName_[id];
   }
+
+  paramObjectFromName(
+    name: string,
+    defaultValue: keyof PartyMemberParamObject,
+    playerNicks: { [name: string]: string },
+  ): PartyMemberParamObject | undefined {
+    const partyMember = this.details.find((member) => member.name === name);
+    if (!partyMember)
+      return;
```

Comment:
This makes me a little worried in the face of the OverlayPlugin bug.  I think this function should never return `undefined`.  It should at least return what it can (name, shortName) and then elide other fields.  And, ideally, if somebody asks for a property that doesn't exist, it should print a warning and pick ~some property rather than nothing.  (Maybe shortName if that exists, and any key after that? Maybe these objects need a `default: string` field in case somebody or some code asks for a non-existent property and we need to fall back on something?)

Doing this would also solve the TODO in the raidboss `test.ts` where you don't get anything if you're not in a party.

Reply:
Also wanted to add that there are some times where triggers might call out to "stack on (non-party member)" for things like hunts, and so in this case even if OP was 100% working for party members, we'd still want to have reasonable fallback behavior.

Reply:
One final question (and maybe there's no good answer here) is what to do if somebody enters `${data.name} (${data.role})` (e.g. #5554) but role is unknown.  I guess it would just say name twice?  Do we need a separate `nameAndRole` entry that could handle this? (Or maybe I am overthinking and this is overkill.)

Reply:
I'm holding off on this piece until I can look at the OP issue in general, but I think the fallback logic is fuzzy and I'm not 100% sure what would be the best behavior here.

Reply:
I think we need this regardless, based on my comment re: hunts.

What options are you considering for fallback logic? Or, what do you see as the issues?

Reply:
In general, what fallback should be used? Should `job`/`role` fall back to `name` or `shortName`, or just display `???`? If we're displaying something like `???`, should that be locale-'d e.g. `Outputs.unknown`, and if so, passing current locale to this location is complicated.

Reply:
We could just fall back to `shortName` always for all objects (and then pick any field after that if it doesn't exist?), but in a general sense where objects are not always players, that seems a little weird.

One thought is that the object should put a (required?) `default` or `fallback` field in that specifies what to be used if a user or code specifies a sub param that doesn't exist.    Personally, I think `shortName` is a reasonable choice for that field.  I think if there's always a valid fallback, then there's no need for `???` or locale plumbing.

Reply:
Coming back to this PR, I still think that this is the missing piece here that prevents me from landing this PR. I feel like there needs to be some story around "what happens if a sub param is specified that doesn't exist". I think in general, it would be better to show something reasonable rather than `???` regardless of the OP party bug.

Reply:
This is updated with the latest commit (also rebased to `main`). Hopefully this makes sense and lines up with what you're looking for?

Reply:
Yes, although I also needed to do e9fcc9690cf24e972e8356c4143278d136a91634 in order for the "no prop specified" case to work.